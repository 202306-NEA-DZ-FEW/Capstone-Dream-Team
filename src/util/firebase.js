// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getFirestore } from "firebase/firestore";
import { getAuth } from "firebase/auth";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
    authDomain: "capstone-project-944c9.firebaseapp.com",
    projectId: "capstone-project-944c9",
    storageBucket: "capstone-project-944c9.appspot.com",
    messagingSenderId: "455661891383",
    appId: "1:455661891383:web:65bbc812478ffda076b4b2",
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const db = getFirestore(app);
export const auth = getAuth(app);
