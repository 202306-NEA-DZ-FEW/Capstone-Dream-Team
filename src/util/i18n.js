import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
