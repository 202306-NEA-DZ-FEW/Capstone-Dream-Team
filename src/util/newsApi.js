export async function fetchData(apiRoute) {
    const url = `https://newsapi.org/v2/${apiRoute}&apiKey=${process.env.NEXT_PUBLIC_NEWS_API}`;
    const response = await fetch(url);
    const data = await response.json();
    return data;
}
