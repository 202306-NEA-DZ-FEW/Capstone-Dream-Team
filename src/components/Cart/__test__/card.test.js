import renderer from "react-test-renderer";

import { db } from "@/util/firebase";

import Card from "../card";

// Mock the Firestore collection and its methods
jest.mock("firebase/firestore", () => ({
    collection: jest.fn(),
}));

// Mock the Firestore instance
jest.mock("@/util/firebase", () => ({
    db: {
        collection: jest.fn(),
    },
}));

jest.mock("next/router", () => ({
    useRouter: jest.fn().mockReturnValue({
        push: jest.fn(),
    }),
}));

jest.mock("next-i18next", () => ({
    useTranslation: jest.fn().mockReturnValue({ t: jest.fn() }),
}));

describe("Card Component", () => {
    const mealObject = {
        quantity: "5",
        name: "baklawa",
        maxMeals: "20",
        price: "2",
        imageUrl: " ",
    };

    it("renders correctly", () => {
        // Mock Firestore behavior
        db.collection.mockReturnValue({
            where: jest.fn(() => ({ get: jest.fn() })),
            add: jest.fn(),
        });

        const tree = renderer.create(<Card mealObject={mealObject} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
