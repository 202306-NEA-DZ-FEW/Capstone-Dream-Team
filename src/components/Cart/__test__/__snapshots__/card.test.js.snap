// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly 1`] = `
<div
  className="flex flex-col p-4 text-lg font-semibold border-b-2 rounded-sm"
>
  <div
    className="flex flex-row gap-3 justify-between "
  >
    <div
      className="flex flex-col md2:flex-row gap-6 items-center justify-center text-center md2:text-left"
    >
      <div
        className="w-28 h-28"
      >
        <img
          alt="image"
          className="w-full h-full"
          src=" "
        />
      </div>
      <div
        className="flex flex-col gap-1 w-36 md:w-60"
      >
        <p
          className="text-lg text-gray-800 font-semibold"
        />
        <p
          className="text-xs text-gray-600 font-semibold hidden md2:block"
        >
          :
           
          <span
            className="font-normal"
          >
            20
          </span>
        </p>
      </div>
    </div>
    <div
      className="self-center text-center w-[100px] hidden md2:block"
    >
      <p
        className="text-gray-800 font-normal text-xl "
      >
        $
        2.00
      </p>
    </div>
    <div
      className="flex flex-row self-center gap-1 w-[100px] justify-center"
    >
      <button
        className="w-5 h-5 self-center rounded-full border border-gray-300"
        onClick={[Function]}
      >
        <svg
          fill="none"
          stroke="#d1d5db"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
        </svg>
      </button>
      <input
        className="w-8 h-8 text-center text-gray-900 text-sm outline-none border border-gray-300 rounded-sm"
        id="baklawa"
        onChange={[Function]}
        type="text"
        value="5"
      />
      <button
        className="w-5 h-5 self-center rounded-full border border-gray-300"
        onClick={[Function]}
      >
        <svg
          fill=""
          stroke="#9ca3af"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 5v14M5 12h14"
          />
        </svg>
      </button>
    </div>
    <div
      className="self-center text-center w-[100px]"
    >
      <p
        className="text-gray-800 font-normal text-xl"
      >
        $
        10.00
      </p>
    </div>
    <div
      className="self-center w-[100px] text-center "
    >
      <button
        className=" "
        onClick={[Function]}
      >
        <svg
          className=""
          height="24px"
          id="Layer_1"
          version="1.1"
          viewBox="0 0 512 512"
          width="24px"
        >
          <g>
            <path
              d="M400,113.3h-80v-20c0-16.2-13.1-29.3-29.3-29.3h-69.5C205.1,64,192,77.1,192,93.3v20h-80V128h21.1l23.6,290.7   c0,16.2,13.1,29.3,29.3,29.3h141c16.2,0,29.3-13.1,29.3-29.3L379.6,128H400V113.3z M206.6,93.3c0-8.1,6.6-14.7,14.6-14.7h69.5   c8.1,0,14.6,6.6,14.6,14.7v20h-98.7V93.3z M341.6,417.9l0,0.4v0.4c0,8.1-6.6,14.7-14.6,14.7H186c-8.1,0-14.6-6.6-14.6-14.7v-0.4   l0-0.4L147.7,128h217.2L341.6,417.9z"
            />
            <g>
              <rect
                height="241"
                width="14"
                x="249"
                y="160"
              />
              <polygon
                points="320,160 305.4,160 294.7,401 309.3,401"
              />
              <polygon
                points="206.5,160 192,160 202.7,401 217.3,401"
              />
            </g>
          </g>
        </svg>
      </button>
    </div>
  </div>
</div>
`;
